src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -1,237 +1,380 @@
                 if not result["OK"]:
                     gLogger.error(result["Message"])
                 elif result["Value"]:
                     oldToken = result["Value"]
                     for tokenType in ["access_token", "refresh_token"]:
                         result = idpObj.revokeToken(oldToken[tokenType], tokenType)
                         if result["OK"]:
-                            gLogger.notice("%s is revoked from" % tokenType, tokenFile)
+                            gLogger.notice(f"{tokenType} is revoked from", self.outputFile)
                         else:
                             gLogger.error(result["Message"])
 
             # Save new tokens to token file
-            result = writeTokenDictToTokenFile(idpObj.token, tokenFile)
+            result = writeTokenDictToTokenFile(idpObj.token, self.outputFile)
             if not result["OK"]:
                 return result
-            tokenFile = result["Value"]
-            gLogger.notice("New token is saved to %s." % tokenFile)
+            self.outputFile = result["Value"]
+            gLogger.notice(f"New token is saved to {self.outputFile}.")
 
             if not DIRAC.gConfig.getValue("/DIRAC/Security/Authorization/issuer"):
                 gLogger.notice("To continue use token you need to add /DIRAC/Security/Authorization/issuer option.")
                 if not self.issuer:
                     DIRAC.exit(1)
                 DIRAC.gConfig.setOptionValue("/DIRAC/Security/Authorization/issuer", self.issuer)
 
-        # Try to get user information
+            # Try to get user authorization information from token
+            result = readTokenFromFile(self.outputFile)
+            if not result["OK"]:
+                return result
+            gLogger.notice(result["Value"].getInfoAsString())
+
+        return S_OK()
+
+    def loginWithCertificate(self):
+        """Login with certificate"""
+        # Search certificate and key
+        if not self.certLoc or not self.keyLoc:
+            cakLoc = Locations.getCertificateAndKeyLocation()
+            if not cakLoc:
+                return S_ERROR("Can't find user certificate and key")
+            self.certLoc = self.certLoc or cakLoc[0]
+            self.keyLoc = self.keyLoc or cakLoc[1]
+        # Generate proxy
+        self.outputFile = self.outputFile or Locations.getDefaultProxyLocation()
+        chain = X509Chain()
+        # Load user cert and key
+        result = chain.loadChainFromFile(self.certLoc)
+        if result["OK"]:
+            result = chain.loadKeyFromFile(self.keyLoc, password=getpass.getpass("Enter Certificate password:"))
+        if not result["OK"]:
+            return result
+
+        # Read user credentials
+        result = chain.getCredentials(withRegistryInfo=False)
+        if not result["OK"]:
+            return result
+        credentials = result["Value"]
+
+        # Remember a clean proxy to then upload it in step 2
```

Comment:
You mean "copy" or "create" or "duplicate"?

Reply:
I mean a shallow copy of `chain` instance, see https://docs.python.org/3/library/copy.html#copy.copy.